### PR TITLE
Initial commit of molecule_openstack_ci role

### PR DESCRIPTION
**Pull request by tehsmyers**

### CODE REVIEW

**Comment by tehsmyers** [comment]
I used my favorite guinea pig, rhsm, to test that this actually works. Surprisingly enough (or not, I suppose?) it did!

I forgot to add the Jenkinsfile, so I'll put that in a new commit.

**Comment by tehsmyers** [comment]
Travis likes the lint, and jenkins likes the play. :)

It's notable that I tested this locally using a different openstack environment than what jenkins is currently configured to use. Since the change in openstack environments precipitated this change, it's heartening to know it works with both environments, as that means we can convert roles to this now and prepare all roles being testing in Jenkins for the pending environment change.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+License
+-------
+
+GPLv3
+
+Author Information
+------------------
+
+Author Name <authoremail@domain.net>
```
Comment (greg-hellings) [inline README.md]: Nice to meet you, Author Name!

File: defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# ssh connection settings
+molecule_openstack_ci_ssh_user: cloud-user
+molecule_openstack_ci_ssh_port: 22
+
+# run_hash will be appended to security group name
+molecule_openstack_ci_security_group_name: molecule
+molecule_openstack_ci_security_group_description:
+  "Security group for testing Molecule"
```
Comment (greg-hellings) [inline defaults/main.yml]: For "testing Molecule" for for "Molecule testing" or for "testing with Molecule"?

File: meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+galaxy_info:
+  author: Sean Myers
+  destiption: Manage the creation and destruction of openstack resources
+              for running molecule in a shared CI environment.
+  description: My Description
+  company: Red Hat, Inc.
+  license: GPLv3
+  min_ansible_version: 2.4
+  platforms:
```
Comment (greg-hellings) [inline meta/main.yml]: Maybe this can be for all types of platforms?

File: molecule/default/destroy.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+- name: Destroy
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  no_log: |-
+    {{ not (lookup('env', 'MOLECULE_DEBUG') | bool or
+       molecule_yml.provisioner.log|default(false) | bool) }}
+  roles:
+    - name: molecule_openstack_ci
```
Comment (greg-hellings) [inline molecule/default/destroy.yml:9]: I love the test-ception going on here!

**Review by greg-hellings — APPROVED** [review]
A few nits, but they don't have to be picked if you don't want to pick them.

File: meta/main.yml (hunk 3 above)
Comment (tehsmyers) [reply meta/main.yml]: I don't know how to specify that in meta/main...I'm not sure it's possible without basically poking galaxy API and generating a list of all the platforms it knows. I think I'll just remove the platforms list entirely (it'll default to an empty list).

I'll also add the "molecule" tag.

File: molecule/default/destroy.yml (hunk 4 above)
Comment (tehsmyers) [reply molecule/default/destroy.yml:9]: This in particular was the part where I expected everything to come crashing down, but...nope. It just worked on the first shot.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      vars:
+        molecule_openstack_ci_state: absent
+```
+
+**Note:** Remember to add "oasis_roles.molecule_openstack_ci" to your
+molecule scenario dependencies.
+
+Molecule scenario `requirements.yml`:
+```yaml
+- name: oasis_roles.molecule_openstack_ci
```
Comment (tehsmyers) [inline README.md:93]: I strongly suspect that this is incomplete, but want to wait until the role is actually on galaxy to be sure. Specifically, though, the dependency step runs after destroy in the normal molecule flow, so the scenarios might have to be tweaked to run dependency before destroy, or we might just want to advise manually running "molecule dependency" before testing, or...something. I want to test it before I document it, but can't really test it until it's up on galaxy.

Comment (tehsmyers) [reply README.md:93]: ...and once tested, I'll have a PR up on meta_skel with the changes.

**Pull request merged**